### ISSUE TITLE
Extra arrows on leader lines

### ISSUE THREAD

**Issue by theoryshaw** [IfcOpenShell/IfcOpenShell#4487]
As outlined:
https://community.osarch.org/discussion/1945/

Can't find a way to replicate, but happens relatively frequently.

**Comment by Moult** [comment]
This only happens in the SVG, right? Can you share a problematic SVG?

**Comment by theoryshaw** [comment]
I think it happens in BB--that is extra vertices get added for some reason. 

As a random example. if you try and print `FLOOR PLAN - ROOF` on the following file, there's a few examples. 
https://hub.openingdesign.com/OpeningDesign/Highland_Haven/src/commit/f7b97cef762a76d16684e1df52b4a510731d038b/Open/Models_and_CAD/BlenderBIM/Highland_Haven.ifc

![image](https://github.com/IfcOpenShell/IfcOpenShell/assets/507113/617f0697-84f9-48db-810c-f2798b430ad2)

**Comment by Moult** [comment]
@aothms is this perhaps related to #887 ?

```python
import ifcopenshell
import ifcopenshell.geom
import ifcopenshell.util.shape

ifc_file = ifcopenshell.open('./Highland_Haven.ifc')
element = ifc_file.by_id(4540642)
for e in ifc_file.traverse(element):
    print(e)

settings = ifcopenshell.geom.settings()
settings.set(settings.INCLUDE_CURVES, True)
settings.set(settings.WELD_VERTICES, True)
shape = ifcopenshell.geom.create_shape(settings, element)

geometry = shape

verts = geometry.verts
print(verts)

edges = geometry.edges
print(edges)

faces = geometry.faces
print(faces)
```

This outputs:

```
#4540642=IfcGeometricCurveSet((#4540641))
#4540641=IfcIndexedPolyCurve(#4540640,$,$)
#4540640=IfcCartesianPointList2D(((1.01334154605865,0.503362596035004),(0.0456077717244625,0.503362596035004),(-6.31875371932983,2.5830409526825)))
(0.3088665032386765, 0.15342491927146923, 0.0, 0.013901248821616197, 0.15342491927146923, 0.0, 0.01390124882161617, 0.15342491927146923, 0.0, -1.9259561336517323, 0.787310882377626, 0.0)
(0, 1, 2, 3)
()
```

i.e. the polycurve should only have 3 verts, but 4 verts are returned. Shouldn't the polycurve remain chained somehow?

**Comment by theoryshaw** [comment]
bump @aothms 
sorry, wanted to cash in my limited bumps. :)

**Comment by theoryshaw** [comment]
I have a feeling this issue is related to the issue with dimension strings flipping upsidesdown too, when new vertices are added... 

https://github.com/user-attachments/assets/d1b1386d-c627-4f8b-9d66-e2f8db3616be

**Comment by Moult** [comment]
Here's the updated test script:

And a file for convenience: 

[foo.ifc.txt](https://github.com/user-attachments/files/19805983/foo.ifc.txt)

```python
import ifcopenshell
import ifcopenshell.geom
import ifcopenshell.util.shape

ifc_file = ifcopenshell.open('/home/dion/foo.ifc')
element = ifc_file.by_id(4540642)
for e in ifc_file.traverse(element):
    print(e)

settings = ifcopenshell.geom.settings()
settings.set("dimensionality", ifcopenshell.ifcopenshell_wrapper.CURVES_SURFACES_AND_SOLIDS)
settings.set("weld-vertices", True)
shape = ifcopenshell.geom.create_shape(settings, element)

geometry = shape

verts = geometry.verts
print(verts)

edges = geometry.edges
print(edges)

faces = geometry.faces
print(faces)
```

**Comment by aothms** [comment]
Should be fixed, not sure if this ever worker before; needed some extra precautions so that the same point was being passed

**Comment by theoryshaw** [comment]
Thanks guys!
Is this where i'm looking to see when these C+ changes get incorporated in the BB releases? 
https://builds.ifcopenshell.org/

**Comment by Moult** [comment]
Yes there's a slight delay, and we only bump when it's not too risky to roll it out and start testing, and when builds are available for all platforms.

**Comment by BIMvoice** [comment]
Verified on a current v0.8.0 build. Reproduced the earlier repro structure synthetically: an `IfcIndexedPolyCurve` on a 3-point `IfcCartesianPointList2D` now returns **3 vertices with chained edges `(0,1),(1,2)`**, not 4 verts with disjoint `(0,1),(2,3)`. Multi-segment, explicit `IfcLineIndex`, `IfcArcIndex`, and closed polycurves all return a single chained path with free ends only at the true endpoints (none for closed loops).

Since Bonsai's leader/line SVG builds its polyline straight from the shape's verts/edges, the extra `marker-mid` arrowheads were a direct consequence of the unwelded interior vertices, and they're gone with the chained output. Looks like @aothms's kernel fix (the "same point being passed" precaution) resolved this. Suggest closing unless anyone can still reproduce extra arrows on a current build.
